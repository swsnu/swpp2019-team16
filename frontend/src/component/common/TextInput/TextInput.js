import React from 'react';
import PropTypes from 'prop-types';
import TextField from '@material-ui/core/TextField';

TextInputMaterial.propTypes = {
  label: PropTypes.string.isRequired,
  defaultValue: PropTypes.string,
  inputProps: PropTypes.object,
  value: PropTypes.string,
  name: PropTypes.string,
  type: PropTypes.string,
  onChange: PropTypes.func,
};

function TextInputMaterial({
  label,
  defaultValue,
  inputProps,
  value,
  name,
  type,
  onChange,
}) {
  return (
    <TextField
      label={label}
      defaultValue={defaultValue}
      inputProps={inputProps}
      value={value}
      name={name}
      type={type}
      onChange={onChange}
      margin="normal"
      variant="outlined"
      fullWidth
    />
  );
}

export default TextInputMaterial;
