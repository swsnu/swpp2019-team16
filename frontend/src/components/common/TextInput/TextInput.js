import React from 'react';
import PropTypes from 'prop-types';
import TextField from '@material-ui/core/TextField';

TextInputMaterial.propTypes = {
  label: PropTypes.string.isRequired,
  defaultValue: PropTypes.string,
  inputProps: PropTypes.object,
  value: PropTypes.string,
  name: PropTypes.string,
  type: PropTypes.string,
  onChange: PropTypes.func,
  margin: PropTypes.string,
  variant: PropTypes.string,
  fullWidth: PropTypes.bool,
};

function TextInputMaterial({
  label,
  defaultValue,
  inputProps,
  value,
  name,
  type,
  onChange,
  margin,
  variant,
  fullWidth,
}) {
  return (
    <TextField
      label={label}
      defaultValue={defaultValue}
      inputProps={inputProps}
      value={value}
      name={name}
      type={type}
      onChange={onChange}
      margin="normal"
      variant="outlined"
      fullWidth={fullWidth}
    />
  );
}

export default TextInputMaterial;
