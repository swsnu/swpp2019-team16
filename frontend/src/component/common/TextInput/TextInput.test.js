import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import TextInput from './';

describe('<TextInput />', () => {
  it('SHOULD match with snapshot', async () => {
    const { container } = render(<TextInput label="TextInput Label" />);
    expect(container).toMatchSnapshot();
  });
});
