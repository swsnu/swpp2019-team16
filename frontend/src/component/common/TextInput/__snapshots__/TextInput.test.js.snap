// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> SHOULD match with snapshot 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink="false"
    >
      TextInput Label
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
